# Task 2

## a

In [ ]:
from pprint import pprint
cards = ["ace", 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
values = [1 if x == "ace" else x for x in cards ]
space = []

for i in range(0, len(cards)):
  for j in range(i+1, len(cards)):
    space.append((cards[i], cards[j], values[i] + values[j]))

pprint(space)

[('ace', 1, 2),
 ('ace', 2, 3),
 ('ace', 3, 4),
 ('ace', 4, 5),
 ('ace', 5, 6),
 ('ace', 6, 7),
 ('ace', 7, 8),
 ('ace', 8, 9),
 ('ace', 9, 10),
 ('ace', 10, 11),
 (1, 2, 3),
 (1, 3, 4),
 (1, 4, 5),
 (1, 5, 6),
 (1, 6, 7),
 (1, 7, 8),
 (1, 8, 9),
 (1, 9, 10),
 (1, 10, 11),
 (2, 3, 5),
 (2, 4, 6),
 (2, 5, 7),
 (2, 6, 8),
 (2, 7, 9),
 (2, 8, 10),
 (2, 9, 11),
 (2, 10, 12),
 (3, 4, 7),
 (3, 5, 8),
 (3, 6, 9),
 (3, 7, 10),
 (3, 8, 11),
 (3, 9, 12),
 (3, 10, 13),
 (4, 5, 9),
 (4, 6, 10),
 (4, 7, 11),
 (4, 8, 12),
 (4, 9, 13),
 (4, 10, 14),
 (5, 6, 11),
 (5, 7, 12),
 (5, 8, 13),
 (5, 9, 14),
 (5, 10, 15),
 (6, 7, 13),
 (6, 8, 14),
 (6, 9, 15),
 (6, 10, 16),
 (7, 8, 15),
 (7, 9, 16),
 (7, 10, 17),
 (8, 9, 17),
 (8, 10, 18),
 (9, 10, 19)]


## b

In [ ]:
taskb = []
for i in range(3, 20):
  print("\nEvent A"+str(i))
  for x in space:
    if x[2] == i:
      print(x)
      taskb.append(x)



Event A3
('ace', 2, 3)
(1, 2, 3)

Event A4
('ace', 3, 4)
(1, 3, 4)

Event A5
('ace', 4, 5)
(1, 4, 5)
(2, 3, 5)

Event A6
('ace', 5, 6)
(1, 5, 6)
(2, 4, 6)

Event A7
('ace', 6, 7)
(1, 6, 7)
(2, 5, 7)
(3, 4, 7)

Event A8
('ace', 7, 8)
(1, 7, 8)
(2, 6, 8)
(3, 5, 8)

Event A9
('ace', 8, 9)
(1, 8, 9)
(2, 7, 9)
(3, 6, 9)
(4, 5, 9)

Event A10
('ace', 9, 10)
(1, 9, 10)
(2, 8, 10)
(3, 7, 10)
(4, 6, 10)

Event A11
('ace', 10, 11)
(1, 10, 11)
(2, 9, 11)
(3, 8, 11)
(4, 7, 11)
(5, 6, 11)

Event A12
(2, 10, 12)
(3, 9, 12)
(4, 8, 12)
(5, 7, 12)

Event A13
(3, 10, 13)
(4, 9, 13)
(5, 8, 13)
(6, 7, 13)

Event A14
(4, 10, 14)
(5, 9, 14)
(6, 8, 14)

Event A15
(5, 10, 15)
(6, 9, 15)
(7, 8, 15)

Event A16
(6, 10, 16)
(7, 9, 16)

Event A17
(7, 10, 17)
(8, 9, 17)

Event A18
(8, 10, 18)

Event A19
(9, 10, 19)


## c

In [ ]:
len(taskb)/len(space)

0.9818181818181818

# Task 4

In [ ]:
candidates = [('m', 18), ('m', 19), ('f', 22), ('f', 20), ('m', 23)]
ok_c = 0
for x in candidates:
  if x[0] == 'f' or x[1] > 21:
    ok_c += 1

print(ok_c / len(candidates)) 

0.6


# Task 14

a. 

East: $98/472=0.207$

North: $85/472=0.18$

South: $99/472=0.209$

West: $190/472=0.403$

b.

$B$ - event of selling a book

$P(B | East) = P(B \cap East) / P(East) = 0.119 / 0.207=0.575$
$P(B | North) = 0.091 / 0.18=0.506$
$P(B | South) = 0.131 / 0.209=0.628$
$P(B | West) = 0.212 / 0.403=0.526$

# Task 20

| | Sucess | Unsuccess |
|--|---|-----|
|Favorable |0.48 | 0.12 |
| Unfavorable | 0.12 | 0.28 |
| Total | **0.6** | **0.4** |

# Task 29

In [8]:
from scipy.special import binom

def p_ret_n(rooms_n, rooms_returned, p_return):
  return binom(rooms_n,rooms_returned) * (p_return**rooms_returned) * ((1-p_return) ** (rooms_n - rooms_returned))

print(str((1-p_ret_n(300, 20, 0.06))*100) + "%")

91.79817694830716%


$\binom{300}{20}0.06^{20}(1-0.06)^{300-20}=0.082 = 8.2\%$ - probability of returning 20 rooms

$1-0.082=0.918=91.8\%$ - at least 280 rooms occupied

# Task 31

In [35]:
from scipy.stats import poisson

print("7 working days")

for i in range(0, 6):
  print(i, poisson.pmf(i,3)*100)

print("\n5 working days")
factor = 3 * (5/7)
for i in range(0, 6):
  print(i, poisson.pmf(i,factor)*100)

7 working days
0 4.978706836786395
1 14.936120510359185
2 22.404180765538776
3 22.404180765538776
4 16.803135574154084
5 10.081881344492457

5 working days
0 11.731916609425078
1 25.139821305910875
2 26.93552282776166
3 19.239659162686902
4 10.306960265725122
5 4.417268685310767


a. $100-4.97=95.03\%$

b. $22.4+22.4+16.8=61.6\%$

c. $25\%$
